src/gengowatcher/browser_debug_launcher.py

```
@@ -0,0 +1,243 @@
+        )
+    except FileNotFoundError as exc:
+        raise RuntimeError(
+            f"Firefox executable not found: {spec.browser_path}"
+        ) from exc
+
+
+async def _probe_firefox_debug_server(debug_url: str) -> bool:
+    try:
+        async with websockets.connect(debug_url, max_size=1_000_000) as websocket:
+            raw_message = await asyncio.wait_for(websocket.recv(), timeout=2)
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)